src/server/game/Entities/Object/Object.cpp

```diff
@@ -1146,9 +1146,13 @@ bool WorldObject::IsWithinLOSInMap(const WorldObject* obj) const
     if (!IsInMap(obj))
         return false;
 
-    float ox, oy, oz;
-    obj->GetPosition(ox, oy, oz);
-    return IsWithinLOS(ox, oy, oz);
+    float x, y, z;
+    if (obj->GetTypeId() == TYPEID_PLAYER)
+        GetPosition(x, y, z);
```

Comment:
Nice catch @Greymane , haven't got the time to test it a lot right now. Will take a loot at this PR next week. Will commit this change ASAP tho :)